dowhy/causal_refuters/bootstrap_refuter.py

```diff
@@ -1,33 +1,75 @@
-    - 'num_simulations': int, CausalRefuter.DEFAULT_NUM_SIMULATIONS by default
-    The number of bootstrap simulations to be run
+    
+    Parameters
+    -----------
+    -'num_simulations': int, CausalRefuter.DEFAULT_NUM_SIMULATIONS by default
+    The number of simulations to be run
     - 'sample_size': int, Size of the original data by default
     The size of each bootstrap sample
+    - 'required_variables': int, list
+    An integer argument means that we select a select number of covariates out of all covariates
+    - 'noise': float, BootstrapRefuter.DEFAULT_STD_DEV by default
+    The standard deviation of the noise to be added to the data
     - 'random_state': int, RandomState, None by default
     The seed value to be added if we wish to repeat the same random behavior. For this purpose, 
     we repeat the same seed in the psuedo-random generator.
     """
 
+    DEFAULT_STD_DEV = 0.1
+    DEFAULT_SUCCESS_PROBABILITY = 0.5
+    DEFAULT_NUMBER_OF_TRIALS = 1
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._num_simulations = kwargs.pop("num_simulations", CausalRefuter.DEFAULT_NUM_SIMULATIONS)
-        self._sample_size = kwargs.pop("sample_size",len(self._data))
-        self._random_state = kwargs.pop("random_state",None)
+        self._num_simulations = kwargs.pop("num_simulations", CausalRefuter.DEFAULT_NUM_SIMULATIONS )
+        self._sample_size = kwargs.pop("sample_size", len(self._data))
+        self._required_variables = kwargs.pop("required_variables", None)
+        self._noise = kwargs.pop("noise", BootstrapRefuter.DEFAULT_STD_DEV )
+        self._random_state = kwargs.pop("random_state", None)
 
         if 'logging_level' in kwargs:
             logging.basicConfig(level=kwargs['logging_level'])
         else:
             logging.basicConfig(level=logging.INFO)
         self.logger = logging.getLogger(__name__)
         
+        # Sanity check the data passed by the user
+        # If the data is invalid, we run the default behavior
+        if self._required_variables is int:
+            if len(self._target_estimand.backdoor_variables) < self._required_variables:
+                self.logger.warning("Too many variables passed.\n The number of backdoor variables is: {}.\n The number of variables passed: {}".format(
+                    len(self._target_estimand.backdoor_variables),
+                    self._required_variables )
+                )
+                self.logger.warning("The bootstrap refuter will follow the default behavior")
+                self._required_variables = None
+
+        elif self._required_variables is list:
+            for variable in self._required_variables:
+                self.logger.warning(variable in self._target_estimand.backdoor_variables), "The variable {} is not not a backdoor variable".format(variable)
+                break
+            self.logger.warning("The bootstrap refuter will follow the default behavior")

```

Comment:
again, better to throw an exception here. but I have a question: what's the problem if the required variables contain a non-backdoor variable? the bootstrap method is simply generating a new dataset, and if the variable is not a confounder, all backdoor methods will ignore it.
I'm thinking that this distinction will be useful later when we extend the refuter for instrumental variable or another method.

Reply:
That makes sense. I can generalize it to check for any variable present in the dataset.